autogain/autogain.py

```diff
@@ -203,22 +219,24 @@ def __init__(self, event, stations):
     def finish(self, reference_nsl, fband, taper):
         for tr in self.traces:
             tr_backup = tr.copy()
-            tr_backup.set_location('B' )
-            tr.ydata -= tr.get_ydata().mean()
+            tr_backup.set_location('B')
+            tr.ydata -= num.int32(tr.get_ydata().mean())  # otherwise dtype error!
```

Comment:
What if the trace's ydata is of type num.float? Can you try if this works?

    tr.ydata -= tr.get_ydata().mean().view(tr.ydata.dtype)

This should (I guess) cast the mean always to the dtype of tr.ydata.

Reply:
You're right, I will try that.

Reply:
That does not work, but this, although it does not look that beautiful
'''
                dtype = type(tr.ydata[0])
                tr.ydata -= dtype(tr.get_ydata().mean())
'''
'''
tr.ydata -= tr.get_ydata().mean().view(tr.ydata.dtype)
'''
does not work, either.